### PR TITLE
Update of graphql definition for using ID fields + character connection.

### PR DESCRIPTION
**Pull request by Vassyli**
Also updated graphql packages to be up-to-date.

### CODE REVIEW

**Comment by austenmc** [comment]
Is this BaseConnection stuff not built in? This feels like boilerplate Relay stuff.

File: src/AppBundle/GraphQL/Mutation/CharacterMutation.php [hunk 1]
```diff
@@ -36,7 +39,8 @@ function createCharacter(string $userId, string $characterName)
         $this->game->getEntityManager()->flush();
 
         return [
-            "character" => new CharacterType($this->getGame(), $character)
+            "characterEdge" => CharacterConnection::createEdgeFor($user, new CharacterType($this->getGame(), $character)),
+            "user" => new UserType($this->getGame(), $user),
```
Comment (austenmc) [inline src/AppBundle/GraphQL/Mutation/CharacterMutation.php:50]: Why return user?

File: src/AppBundle/GraphQL/Resolver/CharacterResolver.php [hunk 2]
```diff
@@ -18,13 +20,42 @@ class CharacterResolver extends BaseManagerService implements ContainerAwareInte
...
+     * Returns a graphql character type from a cursor.
+     * @param type $args Arguments array, as given by CharacterConnection->getEdges()
+     * @return CharacterType
+     */
+    public function getCharacterFromCursor($args = null): CharacterType
+    {
+        $cursor = $args["cursor"];
+        $user = $args["__data"];
+
+        $offset = CharacterConnection::decodeCursor($cursor);
```
Comment (austenmc) [inline src/AppBundle/GraphQL/Resolver/CharacterResolver.php]: Is there error handling needed? If so, let's at least file an issue or put a TODO so we remember to come back.

**Review by austenmc — APPROVED** [review]

File: src/AppBundle/GraphQL/Mutation/CharacterMutation.php (hunk 1 above)
Comment (Vassyli) [reply src/AppBundle/GraphQL/Mutation/CharacterMutation.php:50]: I have not tried to not return the user - I've been looking around for RANGE_ADD examples and they all return the parent field (user) of the connection + the connection edge in the mutation payload.

File: src/AppBundle/GraphQL/Resolver/CharacterResolver.php (hunk 2 above)
Comment (Vassyli) [reply src/AppBundle/GraphQL/Resolver/CharacterResolver.php]: Yes there is, thank you for pointing it out. I will add tests now anyway for both invalid and valid stuff. Incorrect cursor should throw an exception.

**Comment by Vassyli** [comment]
Honestly, I have no clue. There is the Pagination helper, but I couldn't figure out how to use it, thought it's faster to implement it myself. I like this BaseConnection a lot more since it helps to understand the connection between the graphql definition and the class itself.

File: src/AppBundle/GraphQL/Mutation/CharacterMutation.php (hunk 1 above)
Comment (Vassyli) [reply src/AppBundle/GraphQL/Mutation/CharacterMutation.php:50]: Relay asks specifically for this field during the mutation which it probably wouldn't if it isn't required.

**Pull request merged**